### ISSUE TITLE
Enhanced path-alias

### ISSUE THREAD

**Issue by fluffy-critter** [fluffy-critter/Publ#11]
Currently we can map single specific legacy URLs to a specific entry or to a category template. It would also be great if we could map a legacy URL via some higher-level scripting functionality from the templates directly.

On beesbuzz.biz I have this in my `main.py` to do a quick-and-dirty version of this:

```
@app.route('/d/<int:date>.php')
def redirect_date(date):
    return flask.redirect(flask.url_for('category', category='comics', date=date))
```

but it would be great if this could be supported more directly in Publ itself.

**Comment by fluffy-critter** [comment]
Originally I was planning on adding a `Path-Alias-Regex` header but that gets unwieldy fast. What would be better is simply making that above `@app.route` rule something that can be more easily injected at an API level from your application main, for example:

```
@publ.add_path_regex(app, r'/d/([0-9]+)\.php')
def redirect_date(match):
    return flask.url_for('category', category='comics', date=match.group(1))

@publ.add_path_regex(app, r'/d/([0-9]+)_([0-9]+).php')
def redirect_week(match):
    return flask.url_for('category',category='comics',date='{}w{}'.format(match.group(1,2)))
```

**Comment by fluffy-critter** [comment]
Fixed in #114
